### PR TITLE
Newbranch

### PR DESCRIPTION
**Pull request by Yura1977**
Corrected arrays tasks.

### CODE REVIEW

File: ArrayRotation.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        ArrayRotation rotate = new ArrayRotation();
+        int[] arr = { 5, 6, 1, 2, 6 };
+        System.out.println(Arrays.toString(arr) + " - Original Array");
+        arr = rotate.rotateArray(arr, 5, 2);
+        System.out.print("");
+        System.out.println("\n"+ Arrays.toString(arr) + " - Modified Array");
+
+        }
+}
+
```
Comment (BondarenkoJek) [inline ArrayRotation.java]: Delete unnecessary line

File: ArrayRotation.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Java program to rotate an array of length n by
+// d elements
+
+import java.util.Arrays;
+
+public class ArrayRotation {
+
+    /*Function to left rotate arr[] of length n by d */
+    private int[] rotateArray(int[] arr, int n, int d){
```
Comment (BondarenkoJek) [inline ArrayRotation.java]: n - bad name --> length or arrLength or arrayLength.
d --> k 
missed 1 spaces between   ){ --> ) {

File: ArrayRotation.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+        for (int i = 0; i < remainingElem; i++){
+            result[i] = arr[i + d];
+        }
+
+        for (int i = 0; i < d; i++) {
+            result[i + remainingElem] = arr[i];
+        }
+        return result;
+
```
Comment (BondarenkoJek) [inline ArrayRotation.java]: delete unnecessary line

File: ArrayRotation.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    }
+        public static void main(String[] args){
```
Comment (BondarenkoJek) [inline ArrayRotation.java]: Add new line before method 
need horizontal alignment
https://mate-academy.github.io/style-guides/java/java.html#s4-formatting

File: ArrayRotation.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        ArrayRotation rotate = new ArrayRotation();
+        int[] arr = { 5, 6, 1, 2, 6 };
+        System.out.println(Arrays.toString(arr) + " - Original Array");
+        arr = rotate.rotateArray(arr, 5, 2);
+        System.out.print("");
+        System.out.println("\n"+ Arrays.toString(arr) + " - Modified Array");
+
```
Comment (BondarenkoJek) [inline ArrayRotation.java]: delete unnecessary line

File: ArrayRotation.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        }
+        return result;
+
+    }
+        public static void main(String[] args){
+        ArrayRotation rotate = new ArrayRotation();
+        int[] arr = { 5, 6, 1, 2, 6 };
+        System.out.println(Arrays.toString(arr) + " - Original Array");
+        arr = rotate.rotateArray(arr, 5, 2);
+        System.out.print("");
```
Comment (BondarenkoJek) [inline ArrayRotation.java]: System.out.print(""); delete

File: Ranges.java [hunk 7]
```diff
@@ -1,34 +1,42 @@
...
-        }
-        return result.toString();
-    }
+import java.util.Arrays;
+
+
+public class Ranges {
+    public static void main(String[] args) {
+
+        int array[] = { 1,2,3,5,8,9,10,13,14,15,16 };//1 2 3 5 8 9 10 13 14 15 16
```
Comment (BondarenkoJek) [inline Ranges.java]: int array[] = { 1,2,3,5,8,9,10,13,14,15,16 }; --> int array[] = {1, 2, 3, 5, 8, 9, 10, 13, 14, 15, 16};
https://mate-academy.github.io/style-guides/java/java.html#s4.8.3-arrays

File: Ranges.java [hunk 8]
```diff
@@ -1,34 +1,42 @@
...
-            result.append("[").append(array[array.length - 1]).append("]");
-        }
-        return result.toString();
-    }
+import java.util.Arrays;
+
+
+public class Ranges {
+    public static void main(String[] args) {
+
```
Comment (BondarenkoJek) [inline Ranges.java]: delete unnecessary line

File: Ranges.java [hunk 9]
```diff
@@ -1,34 +1,42 @@
...
+        int array[] = { 1,2,3,5,8,9,10,13,14,15,16 };//1 2 3 5 8 9 10 13 14 15 16
+        System.out.println(Arrays.toString(array) + " - Original Array");
+        System.out.println(arrayRanges(array) + " - New Array");
+}
+
+    public static <overwrite> String arrayRanges(int[] array) {
```
Comment (BondarenkoJek) [inline Ranges.java]: overwrite --???
delete

File: Ranges.java [hunk 10]
```diff
@@ -1,34 +1,42 @@
...
+        StringBuilder result = new StringBuilder();
+        int minValue = array[0];
+        int maxValue;
+        for (int i = 1; i < array.length; i++) {
+            if (array[i] - array[i - 1] > 1) {
+                maxValue = array[i - 1];
+                if (maxValue == minValue) {
+                    result.append("[").append(minValue).append("]");
+                } else {
+                    result.append("[").append(minValue).append(" ").append(maxValue).append("]");
```
Comment (BondarenkoJek) [inline Ranges.java]: you have method which do it
	                    result.append("[").append(minValue).append("]");
	                } else {
	                    result.append("[").append(minValue).append(" ").append(maxValue).append("]");

File: Ranges.java [hunk 11]
```diff
@@ -1,34 +1,42 @@
...
+                minValue = array[i];
+            }
+        }
+        boundary(result, minValue, array[array.length - 1]);
+
+        return result.toString();
+    }
+
+    public static void boundary (StringBuilder sb, int minValue, int maxValue){
+
```
Comment (BondarenkoJek) [inline Ranges.java]: delete unnecessary line

File: Ranges.java [hunk 12]
```diff
@@ -1,34 +1,42 @@
...
+        sb.append("[").append(minValue);
+        if (maxValue != minValue) {
+            sb.append(" ").append(maxValue);
+        }
+        sb.append("]");
+
+    }
+
```
Comment (BondarenkoJek) [inline Ranges.java]: between braces should not be spaces 
after last braces should be 1 spaces

File: RestoreArray.java [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import java.util.Arrays;
+
+public class RestoreArray {
+
```
Comment (BondarenkoJek) [inline RestoreArray.java]: delete unnecessary line

File: ArrayRotation.java [hunk 14]
```diff
@@ -0,0 +1,29 @@
+// Java program to rotate an array of length n by
+// k elements
+
+import java.util.Arrays;
+
+public class ArrayRotation {
+    /*Function to left rotate arr[] of length n by k */
+    private int[] rotateArray(int[] arr, int n, int k) {
+        int[] result = new int[arr.length];
```
Comment (BondarenkoJek) [inline ArrayRotation.java]: way you write new int[arr.length]; if you have n -???
you should rename var n --> length will be better

File: Ranges.java [hunk 15]
```diff
@@ -1,34 +1,35 @@
...
+public class Ranges {
+    public static void main(String[] args) {
+        int array[] = {
+                1, 2, 3, 5, 8, 9, 10, 13, 14, 15, 16
+        };
+        System.out.println(Arrays.toString(array) + " - Original Array");
+        System.out.println(arrayRanges(array) + " - New Array");
+    }
+
+    public static String arrayRanges(int[] array) {
```
Comment (BondarenkoJek) [inline Ranges.java]: Method names are typically verbs or verb phrases. For example, sendMessage or stop.
https://mate-academy.github.io/style-guides/java/java.html#s5.2.3-method-names

File: Ranges.java [hunk 16]
```diff
@@ -1,34 +1,35 @@
...
+                maxValue = array[i - 1];
+                boundary(result, minValue, maxValue);
+                minValue = array[i];
+            }
+        }
+        boundary(result, minValue, array[array.length - 1]);
+        return result.toString();
+    }
+
+    public static void boundary(StringBuilder sb, int minValue, int maxValue) {
```
Comment (BondarenkoJek) [inline Ranges.java]: Method names are typically verbs or verb phrases. For example, sendMessage or stop.
https://mate-academy.github.io/style-guides/java/java.html#s5.2.3-method-names

**Comment by BondarenkoJek** [comment]
Should create package for classes

File: RestoreArray.java [hunk 17]
```diff
@@ -0,0 +1,22 @@
+import java.util.Arrays;
+
+public class RestoreArray {
+    static int[] modifyArray(int[] arr) {
+        for (int i = 0; i < arr.length; i++) {
+            if (arr[i] < 0 && arr[i - 1] < arr[i + 1]) {
+                arr[i] = arr[i - 1] + 1;
```
Comment (BondarenkoJek) [inline RestoreArray.java]: this method actually returns the same array it receives as an argument

Comment (Yura1977) [reply RestoreArray.java]: I run everything works

[5, 6, 1, 2, 6] - Original Array

[1, 2, 6, 5, 6] - Modified Array

File: ArrayRotation.java (hunk 14 above)
Comment (Yura1977) [reply ArrayRotation.java]: I already changed n to length

File: Ranges.java [hunk 18]
```diff
@@ -1,34 +1,35 @@
...
-            result.append("[").append(minValue).append(" ").append(array[array.length - 1]).append("]");
-        } else {
-            result.append("[").append(array[array.length - 1]).append("]");
-        }
-        return result.toString();
-    }
-}
\ No newline at end of file
+import java.util.Arrays;
+
```
[the remark this answers is not in the document]
Comment (Yura1977) [reply Ranges.java]: Corrected

File: RestoreArray.java (hunk 17 above)
Comment (BondarenkoJek) [reply RestoreArray.java]: try to print original array after invoke method

**Pull request merged**